### PR TITLE
ontag_k3s_certification job flaky test fixes

### PR DESCRIPTION
**Pull request by vivek-infracloud**
## Issue: 

 There are many flaky tests in jenkins job rancher-v3_ontag_import_k3s_certification
Latest run show following tests that fails despite everything is fine in the cluster.

tests.v3_api.test_service_discovery.test_dns_record_type_workload_when_workload_scale_down
tests.v3_api.test_service_discovery.test_dns_record_type_multiple_workloads
tests.v3_api.test_websocket.test_websocket_view_logs
tests.v3_api.test_service_discovery.test_dns_record_type_hostname
tests.v3_api.test_service_discovery.test_service_discovery_when_workload_upgrade
tests.v3_api.test_service_discovery.test_dns_record_type_workload_when_workload_scale_up
tests.v3_api.test_service_discovery.test_dns_record_type_workload_when_workload_upgrade
tests.v3_api.test_websocket.test_websocket_launch_kubectl
tests.v3_api.test_workload.test_wl_with_clusterIp
tests.v3_api.test_workload.test_wl_with_clusterIp_scale_and_upgrade
tests.v3_api.test_network_policy.test_connectivity_between_pods
## Problem

 These tests fail due to timing issue or failure to reach pod in first attempt.
Earlier PR that I raised for fixing the same flaky tests resulted into more flaky tests in test_ingress.py hence I closed that PR and raised this one.
## Solution

 This PR serves the purpose to fix this issue. I made sure that there are no failures in test_ingress.py with the current changes.

### CODE REVIEW

**Comment by izaac** [comment]
Needs squash new reviews and a freefrom run.

**Comment by slickwarren** [comment]
it looks like this is still a work in progress, and the commit is still not squashed. Could you please mark as draft until you're ready for a review?

File: tests/validation/tests/v3_api/common.py [hunk 1]
```diff
@@ -1031,16 +1045,8 @@ def validate_dns_entry(pod, host, expected, port=TEST_IMAGE_PORT, retry_count=3)
...
-    cmd_output = kubectl_pod_exec(pod, cmd)
-
-    if str(pod.name) not in str(cmd_output):
-        for i in range(0, retry_count):
-            cmd_output = kubectl_pod_exec(pod, cmd)
-            if str(pod.name) in str(cmd_output):
-                break
-            else:
-                time.sleep(5)
+        cmd = 'ping -c 2 -W 2 {0}'.format(host)
```
Comment (izaac) [inline tests/validation/tests/v3_api/common.py:1048]: If i understand correctly we're delegating the retry and validation in `retry_cmd` now so I would say we should be more clear in the method name at what's doing. Like the `retry_cmd` to `retry_cmd_validate_expected` (this is just a naming idea, not a request).

**Review by izaac — CHANGES_REQUESTED** [review]
Just a question and a method naming suggestion.

File: tests/validation/tests/v3_api/common.py [hunk 2]
```diff
@@ -1018,9 +1031,10 @@ def retry_dig(host, pod, expected, retry_count=3):
         expected_length = len(expected)
         if dig_length >= expected_length:
             return dig_output
-        elif dig_length < expected_length:
-            time.sleep(3)
-    pytest.fail(f"failed to get the expected number of dns hosts from dig")
+        start += +5
```
[the remark this answers is not in the document]
Comment (markusewalker) [reply tests/validation/tests/v3_api/common.py:1034]: Maybe this is supposed to be `start += 5`? Not sure what the `+5` is either...

File: tests/validation/tests/v3_api/common.py (hunk 1 above)
Comment (vivek-infracloud) [reply tests/validation/tests/v3_api/common.py:1048]: Right, that makes more sense. I changed the `retry_cmd` to `retry_cmd_validate_expected`.

File: tests/validation/tests/v3_api/common.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vivek-infracloud) [reply tests/validation/tests/v3_api/common.py]: Ohh, that's typo. I corrected it.

**Review by markusewalker — APPROVED** [review]

**Pull request merged**